diff --git a/kernel-open/nvidia/os-interface.c b/kernel-open/nvidia/os-interface.c
--- a/kernel-open/nvidia/os-interface.c
+++ b/kernel-open/nvidia/os-interface.c
@@ -40,5 +40,6 @@
 #include <linux/pid.h>
 #include <linux/pid_namespace.h>
+#include <linux/string.h>
 #if defined(CONFIG_LOCKDEP)
 #include <linux/lockdep.h>
 #endif // CONFIG_LOCKDEP
@@ -764,2 +765 @@
-    strncpy(buf, current->comm, len - 1);
-    buf[len - 1] = '\0';
+    strscpy(buf, current->comm, len);
